### PR TITLE
Feature/ubi 343/auth dev

### PR DESCRIPTION
**Pull request by JeremiSavard**
#### What changes does this PR contain?
SignIn and SignUp to keycloak 

#### Which JIRA ticket ID's are associated with this PR?
ubi-343

#### Does this PR change the outward facing API?
no

#### Additional comments and information
integration tests are wip
@ubivius/microservices

### CODE REVIEW

File: pkg/handlers/post.go [hunk 1]
```diff
@@ -1,16 +1,175 @@
...
+// SignIn will fetch the acces token from Keycloak and return user data
+func (authHandler *AuthHandler) SignIn(responseWriter http.ResponseWriter, request *http.Request) {
+	signinPath := "http://localhost:8080/auth/realms/ubivius/protocol/openid-connect/token"
+
+	requestbody, _ := ioutil.ReadAll(request.Body)
+
+	username := ExtractValue(string(requestbody), "username")
+	password := ExtractValue(string(requestbody), "password")
+	log.Println(username)
+	log.Println(password)
```
Comment (MaximeStOnge) [inline pkg/handlers/post.go:22]: On va devoir ne pas log le password

File: pkg/handlers/post.go [hunk 2]
```diff
@@ -1,16 +1,175 @@
...
+
+// SignUp will register a new user in keycloak and in our user database
+func (authHandler *AuthHandler) SignUp(responseWriter http.ResponseWriter, request *http.Request) {
+	newUserPath := "http://localhost:8080/auth/admin/realms/ubivius/users"
+
+	requestbody, _ := ioutil.ReadAll(request.Body)
+	username := ExtractValue(string(requestbody), "username")
+	firstName := ExtractValue(string(requestbody), "firstName")
+	lastName := ExtractValue(string(requestbody), "lastName")
+	email := ExtractValue(string(requestbody), "email")
```
Comment (MaximeStOnge) [inline pkg/handlers/post.go:91]: Where is the password?

**Review by MaximeStOnge — CHANGES_REQUESTED** [review]

File: pkg/handlers/post.go (hunk 2 above)
Comment (JeremiSavard) [reply pkg/handlers/post.go:90]: j'ai ajouté la partie pour set le password

File: pkg/handlers/auth.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+// returns - the value for the given key
+func ExtractValue(body string, key string) string {
+	keystr := "\"" + key + "\":[^,;\\]}]*"
+	r, _ := regexp.Compile(keystr)
+	match := r.FindString(body)
+	keyValMatch := strings.Split(match, ":")
+	return strings.ReplaceAll(keyValMatch[1], "\"", "")
+}
+
+//Not used yet, but could be usefull
```
[the remark this answers is not in the document]
Comment (JeremiSavard) [reply pkg/handlers/auth.go:62]: I ended up using it in the sign in

**Review by ArcticWalrus — DISMISSED** [review]

**Review by MaximeStOnge — APPROVED** [review]

**Review by ArcticWalrus — APPROVED** [review]

**Pull request merged**